src/control/tools/EditSelectionContents.h

```diff
@@ -173,6 +175,11 @@ class EditSelectionContents : public ElementContainer, public Serializeable
 	 */
 	vector<Element*> selected;
 
```

Comment:
maybe use a std::deque, since it has a better performance when the maximum size of the vector is unknown, it allows also an insertion of elements onto the front with O(1) (if the order of the for-loop must be inverted this can be used to invert it and still use a `for(auto&&element:elements)`).

Reply:
I'm not sure why vector doesn't work here... The .rend and .rbegin iterators are both O(1) and getting the next is O(1) so I think the whole iteration cost (either direction) is O(n). 

But I do understand if it's better to restrict functionality to only the functionality we need right now, aka queueing and iterating over contents of the queue.

Reply:
Of course a classical std::vector works here, but it will reallocate on element insertion (insert/emplace/emplace_back/push_back), that results in the fact, that a vector is on average slower than a std::deque. Also you can't insert Elements in front of a std::vector without moving/coping all elements, with a std::deque you can do that. For further informations please read https://en.cppreference.com/w/cpp/container/deque .